### DISCUSSION TITLE
part-ii/2020-77/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#172] [Question comments]
Part II 2020 Graph Theory: Paper 3, Section II, $17 \mathrm{G}$ 

https://questions.tripos.org/part-ii/2020-77/

**Comment by stevezshen** [comment]
<details>
<summary>Spoiler warning</summary>

iv) If every edge lies in a triangle, done. Otherwise $uv$ is an edge that is contained in no triangles. Consider $H = G-u-v$. If it has more than $(n-1)^2+1$ edges, this happens when $d(u)+d(v)<2n$, then delete an edge from a triangle in $H$ to form $H'$, and apply induction hypothesis to $H'$. If $d(u)+d(v)=2n$, then either $u$ or $v$ is in a triangle. 

</details>
